src/rgw/rgw_op.cc

```diff
@@ -4473,18 +4471,25 @@ void RGWBulkDelete::pre_exec()
 
 void RGWBulkDelete::execute()
 {
-  deleter = std::unique_ptr<Deleter>(new Deleter(store, s));
+  try {
+    deleter = std::unique_ptr<Deleter>(\
+          new RGWBulkDelete::Deleter(store, s));
+  } catch (std::bad_alloc) {
+    op_ret = -ENOMEM;
+    return;
+  }
```

Comment:
i'm not convinced that catching allocation failures here is a good idea. if we return -ENOMEM at this point, are we actually able to send a reply back to the client without any other allocations? this is an error path that we don't have a way to test. once we're out of memory, there's little we can do to recover gracefully so it's generally best to crash early